sdk/appconfiguration/azure-spring-cloud-appconfiguration-config/src/main/java/com/azure/spring/cloud/config/implementation/ConfigurationClientWrapper.java

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.spring.cloud.config.implementation;
+
+import java.time.Instant;
+
+import com.azure.core.exception.HttpResponseException;
+import com.azure.core.http.rest.PagedIterable;
+import com.azure.data.appconfiguration.ConfigurationClient;
+import com.azure.data.appconfiguration.models.ConfigurationSetting;
+import com.azure.data.appconfiguration.models.SettingSelector;
+import com.azure.spring.cloud.config.NormalizeNull;
+
+/**
+ * Wrapper for Configuration Client to manage backoff.
+ */
+class ConfigurationClientWrapper {
```

Comment:
Just a thought, should we define an interface like 
```java 

interface GeoRedundantClient<T> {

    T getClient();
    
    int getFailedAttempts();
    
    Instant getBackoffEndTime();
    
    void updateBackoffEndTime(Instant backoffEndTime);
}

```
?

Reply:
I am not sure if this is needed. I don't see us supporting multiple types of clients for connecting to App Configuration.